### ISSUE TITLE
[Reports] Remind me of query behind rn=emailopenratio?

### ISSUE THREAD

**Issue by daver78** [coa0329/BhaangWeb#227]
@amalekshahi, I want to use the data provided in `rn=emailopenratio` within DV.  Can you remind me of how you constructed the query for this?

You can either paste the query or describe it in English.

Lastly, it runs fairly slowly; do you see a way to speed it up?  If not, we may need to cache it. 

Here's a query you can try: `https://studio.dashboard.mdl.io//api/Report/GetAdminReportView?rn=emailopenratio&si=0&ai=679&st=0&fd=8/8/2017&td=11/6/2017&seed=False&menu=True&token=3230c79f-0003-4a0e-9a34-bcbc5f1d7b6e&pb=true&filter=False&title=MindFire%20Master`

![image](https://user-images.githubusercontent.com/7561473/32448974-39aba6b2-c2c5-11e7-8316-4ee3acd53713.png)

**Comment by amalekshahi** [comment]
The actual query is below. Basically, it aggregates Opens, sent and ratio (percentage of opens to sent) across days of the week (Sun .. Sat) for an account and for a date range.
We went through different revisions of this query and the current one is the most efficient one I could develop, given the complexity of the query and the exclude clauses:

Select DayOfWeekNumber, DayOfWeek, Sum (Opened), Sum (Sent), Avg (case when Sent = 0 then 0 else 100*Opened/ Sent end) as Ratio From
		(select DayOfWeekNumber, DayOfWeek, et.OutboundSchedule_ID
				, Count (Distinct (CASE WHEN et.Event_ID = 20103 THEN et.Contact_ID ELSE null END)) as Opened
				, Sum (CASE WHEN et.Event_ID = 20101 and os.TriggerContact_ID is null and et.ContactsCount > 100 THEN et.ContactsCount ELSE 0 END) as Sent
		from 
			foxtrot.EventTransaction et with (nolock)
			join foxtrot.dim_date D with (nolock) on d.ID = et.dim_date_ID
			left outer join foxtrot.OutboundSchedule os  with (nolock) on et.account_ID = os.account_ID and et.OutboundSchedule_ID = os.ID
		where 
			et.Account_ID = 679
			and et.ServiceType_ID = 201
			and d.date >= '8/8/2017' and d.date <= '11/6/2017'
		group by d.DayOfWeek, DayOfWeekNumber, et.OutboundSchedule_ID
		having 
			Sum (CASE WHEN et.Event_ID = 20101 and os.TriggerContact_ID is null and et.ContactsCount > 100 THEN et.ContactsCount ELSE 0 END) > 0 and 
			Count (Distinct (CASE WHEN et.Event_ID = 20103 THEN et.Contact_ID ELSE null END)) > 0 and 
			Sum (CASE WHEN et.Event_ID = 20101 and os.TriggerContact_ID is null and et.ContactsCount > 100 THEN et.ContactsCount ELSE 0 END) > Count (Distinct (CASE WHEN et.Event_ID = 20103 THEN et.Contact_ID ELSE null END))
		--order by DayOfWeekNumber, d.DayOfWeek
		) OBSHour
group by DayOfWeekNumber, DayOfWeek
Order by DayOfWeekNumber

**Comment by daver78** [comment]
@amalekshahi OK, now I am remembering the details a bit more.  I didn't dissect your SQL yet, but are we looking at the open rate for emails sent the SAME day?

**Comment by amalekshahi** [comment]
No, not necessary. I aggregate sent per weekday and open per weekday separately. Opens might be for Sent for the same day or previous days
